src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -71,7 +72,8 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 {
                     var handler = ({{endpoint.EmitHandlerDelegateCast()}})del;
                     EndpointFilterDelegate? filteredInvocation = null;
-
+                    var serviceProvider = options?.ServiceProvider ?? options?.EndpointBuilder?.ApplicationServices;
```

Comment:
Nit: Move this into `EmitJsonPreparation` since it is the only codepath that needs it. Otherwise, I believe we might be getting CS0219 warnings in the generated code...